lib/puppet/util/log.rb

```diff
@@ -170,6 +170,17 @@ def Log.with_destination(destination, &block)
   def Log.newmessage(msg)
     return if @levels.index(msg.level) < @loglevel
 
+    if msg.message.valid_encoding?
```

Comment:
In this case should we also bump up the `level`?

Our levels are:
* :debug
* :notice
* :info
* :warning
* :err